tests/test_variants.py

```diff
@@ -35,38 +35,76 @@ def test_later_spec_priority(single_version, no_numpy_version):
     assert len(combined_spec["python"]) == 2
 
 
-def test_get_package_variants_from_file(testing_workdir, testing_config, no_numpy_version):
-    with open('variant_example.yaml', 'w') as f:
-        yaml.dump(no_numpy_version, f, default_flow_style=False)
-    testing_config.variant_config_files = [os.path.join(testing_workdir, 'variant_example.yaml')]
+def test_get_package_variants_from_file(
+    testing_workdir, testing_config, no_numpy_version
+):
+    variants_path = Path(testing_workdir, "variant_example.yaml")
+    variants_path.write_text(yaml.dump(no_numpy_version, default_flow_style=False))
+
+    testing_config.variant_config_files = [str(variants_path)]
     testing_config.ignore_system_config = True
-    metadata = api.render(os.path.join(thisdir, "variant_recipe"),
-                            no_download_source=False, config=testing_config)
+
+    metadata = api.render(
+        os.path.join(variants_dir, "variant_recipe"),
+        no_download_source=False,
+        config=testing_config,
+    )
+
     # one for each Python version.  Numpy is not strictly pinned and should present only 1 dimension
     assert len(metadata) == 2
-    assert sum('python >=2.7,<2.8' in req for (m, _, _) in metadata
-               for req in m.meta['requirements']['run']) == 1
-    assert sum('python >=3.5,<3.6' in req for (m, _, _) in metadata
-               for req in m.meta['requirements']['run']) == 1
+    assert (
+        sum(
+            "python >=2.7,<2.8" in req
```

Comment:
do we update the python version here??

Reply:
not necessary, see the `no_numpy_version` fixture, which defines the variants we are building against

Reply:
sorry that was wrong comment.. I'm deleting it

Reply:
The versions of Python utilized here are insignificant, but I went ahead and refactored this test (and other similar tests) to make this less ambiguous.